modules/core/kivent_core/systems/gameview.pyx

```diff
@@ -85,13 +92,38 @@ cdef class GameView(GameSystem):
     window_size = ListProperty((100., 100.))
     touch_pass_through = BooleanProperty(False)
 
+    _camera_rotate = NumericProperty(0)
+
+    def _get_rotate(self):
+        return self._camera_rotate
+
+    def _set_rotate(self, value):
+        # Adjust camera_pos so that rotation is about screen center
+        camera_pos = self.camera_pos
+        prev_value = self._camera_rotate
+        camera_size = self.size
+        camera_scale = self.camera_scale
+
+        screen_center = (camera_size[0] * camera_scale * 0.5,
+                         camera_size[1] * camera_scale * 0.5)
+        screen_center_rotated = self._rotate_point(screen_center, value - prev_value)
+
+        self.camera_pos = (camera_pos[0] + screen_center_rotated[0] - screen_center[0],

```

Comment:
We should not be setting camera_pos here. Instead in the update_render_state, the center and screen_center_rotated should be calculated and applied only for that frame.